public/app/plugins/panel/geomap/utils/measure.tsx

```diff
@@ -0,0 +1,159 @@
+import { Geometry } from 'ol/geom';
+import { getArea, getLength } from 'ol/sphere';
+import React from 'react';
+
+import { FormattedValue, formattedValueToString, getValueFormat, SelectableValue, toFixedUnit } from '@grafana/data';
+
+import { CustomSelectOption } from '../components/CustomSelectOption';
+
+type MeasureAction = 'area' | 'length';
+
+export interface MapMeasureOptions {
+  action: MeasureAction;
+  unit: string;
+  unitLabel: string;
+  color?: string;
+}
+
+export function getMapMeasurement(opts: MapMeasureOptions, geo: Geometry): string {
+  let v = 0;
+  let action = measures[0];
+  if (opts.action === 'area') {
+    action = measures[1];
+    v = getArea(geo);
+  } else {
+    v = getLength(geo);
+  }
+  return formattedValueToString(action.getUnit(opts.unit).format(v));
+}
+
+export interface MapMeasure extends SelectableValue<MeasureAction> {
+  geometry: 'Polygon' | 'LineString';
+  units: MapUnit[];
+
+  /**
+   * This will get the best unit for the selected string or a default
+   * This is helpful when converting from area<>length, we should try to stay
+   * in the same category of unit if possible
+   */
+  getUnit: (v?: string) => MapUnit;
+}
+
+export interface MapUnit extends SelectableValue<string> {
+  format: (si: number) => FormattedValue;
+}
+
+//const option: typeof CustomSelectOption;
+
+export const measures: MapMeasure[] = [
+  {
+    value: 'area',
+    label: 'Area',
+    geometry: 'Polygon',
+    units: [
+      // { name: 'Square Meters (m²)', id: 'areaM2', fn: toFixedUnit('m²') },
+      // { name: 'Square Feet (ft²)', id: 'areaF2', fn: toFixedUnit('ft²') },
+      // { name: 'Square Miles (mi²)', id: 'areaMI2', fn: toFixedUnit('mi²') },
+
+      {
+        component: () => {
+          return <CustomSelectOption value="Square Meters (m²)" />;
```

Comment:
lets keep react out of the utils function :)

Also why necessary?  can we just use the label?

Reply:
I went this route so we can have control over the styling, in this case font size, but in the future maybe more. I feel the default select OPTIONS font is too big for this application. Are you aware of any way to tweak this without a custom component?